python/nvstrings.py

```diff
@@ -878,11 +878,10 @@ def cat(self, others=None, sep=None, na_rep=None):
 
         Parameters
         ----------
-        others : List of str
+        others : nvstrings or list of nvstrings
             Strings to be appended.
```

Comment:
Let's raise an informative error if `others` is a Python list of CPU strings to avoid segfaulting.

Reply:
This is passing cuDF string pytests for `str.cat`, which includes handling a list of Python strings. Pending this safeguard, this LGTM David

Reply:
Argh, yes. Good catch.